scripts/upgrade

```diff
@@ -98,124 +107,100 @@ then
 # Check if the repo can be updated with git
   if [ `cd $final_path && git rev-parse --is-inside-work-tree` ]; 
   then
-    # Update through git
+    # Update through Git
 	pushd "$final_path"
       git fetch
       git checkout stable
       git pull
+      git reset --hard $version_commit
 	popd
 	pushd "$final_path/addon"
       git fetch
       git checkout stable
       git pull
+      git reset --hard $addons_version_commit
 	popd
-# If git is not present upgrade through manual method 
+# If Git is not present upgrade through manual method 
 else
 
-    # Create a temporary directory and backup smarty3 folder
-    tmpdir="$(ynh_smart_mktemp 6000)"
-    ynh_script_progression --message="Upgrading source files..."
-    mkdir -p "$tmpdir/view/"
-    cp -a "$final_path/view/smarty3" "$tmpdir/view/smarty3"
+  # Create a temporary directory and backup smarty3 folder
+	tmpdir="$(mktemp -d)"
+	ynh_script_progression --message="Upgrading source files..." --weight=3
```

Comment:
Maybe to move status message somewhere to line 106 (before the git upgrade path too), or to create another status message for git pulls